File: 0001_Two_Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        HashMap<Integer, Integer> targets = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (targets.containsKey(nums[i])) {
                result[1] = i;
                result[0] = targets.get(nums[i]);
                break;
            }
            targets.put(target - nums[i], i);
        }
        return result;
    }
}

File: 0001_Two_Sum.go
package leetcode

func twoSum(nums []int, target int) []int {
	hash := make(map[int]int, len(nums))

	for i, n := range nums {
		if j, ok := hash[n]; ok {
			return []int{j, i}
		}
		hash[target-n] = i
	}

	return nil
}
